src/main/java/org/codeforamerica/messaging/services/MessageService.java

```diff
@@ -17,34 +20,48 @@ public class MessageService {
     private final SmsService smsService;
     private final EmailService emailService;
     private final MessageRepository messageRepository;
+    private final JobRequestScheduler jobRequestScheduler;
 
-
-    public MessageService(SmsService smsService, EmailService emailService, MessageRepository messageRepository) {
+    public MessageService(SmsService smsService, EmailService emailService, MessageRepository messageRepository,
+            JobRequestScheduler jobRequestScheduler) {
         this.smsService = smsService;
         this.emailService = emailService;
         this.messageRepository = messageRepository;
+        this.jobRequestScheduler = jobRequestScheduler;
+    }
+
+    public Message scheduleMessage(MessageRequest messageRequest) {
+        Message message = saveMessage(messageRequest);
+
+        OffsetDateTime sendAt = messageRequest.getSendAt() == null ? OffsetDateTime.now() : messageRequest.getSendAt();
+        JobId id = jobRequestScheduler.schedule(sendAt, new SendMessageJobRequest(message.getId()));
+        log.info("Scheduled job {} to run at {}", id, sendAt);
+        return message;
     }
 
-    public Message sendMessage(MessageRequest messageRequest) {
+    public Message saveMessage(MessageRequest messageRequest) {
         Message message = Message.builder()
                 .subject(messageRequest.getSubject())
                 .body(messageRequest.getBody())
                 .toPhone(messageRequest.getToPhone())
                 .toEmail(messageRequest.getToEmail())
                 .build();
-        SmsMessage sentSmsMessage;
-        EmailMessage sentEmailMessage;
-        if (message.getToPhone() != null) {
-            sentSmsMessage = this.smsService.sendSmsMessage(message.getToPhone(), message.getBody());
+        messageRepository.save(message);
+        return message;
+    }
+
+    public void sendMessage(Long messageId) {
+        log.info("Sending message #{}", messageId);
+        Message message = messageRepository.findById(messageId).get();
```

Comment:
I noticed that IntelliJ warns: `'Optional.get()' without 'isPresent()' check `. I know that as it is currently implemented this will never be an issue, but it would be nice for this to be safer.

Curious why only the id is passed through instead of the `message`, though I do remember there being an issue with something similar in the prototype?

Reply:
I'll fix it to log and throw an exception.

Only the `id` is passed based on https://www.jobrunr.io/en/documentation/background-methods/best-practices/.